src/blueapi/worker/reworker.py

```diff
@@ -64,20 +69,45 @@ def __init__(
         self._data_events = EventPublisher()
         self._status_lock = RLock()
         self._status_snapshot = {}
+        self._started = Event()
+        self._stopping = Event()
+        self._stopped = Event()
 
     def submit_task(self, name: str, task: Task) -> None:
         active_task = ActiveTask(name, task)
         LOGGER.info(f"Submitting: {active_task}")
         self._task_queue.put(active_task)
 
-    def run_forever(self) -> None:
+    def start(self) -> None:
+        if self._started.is_set():
+            raise Exception("Worker is already running")
+        run_worker_in_own_thread(self)
+
+    def stop(self) -> None:
+        LOGGER.info("Attempting to stop worker")
+
+        # If the worker has not yet started there is nothing to do.
+        if self._started.is_set():
+            self._task_queue.put(KillSignal())
+            self._stopped.wait(timeout=30.0)
```

Comment:
should we make this timeout some global var like BLUESKY_TIMEOUT that is set elsewhere?